spyderlib/widgets/sourcecode/codeeditor.py

```diff
@@ -2654,6 +2642,22 @@ def keyPressEvent(self, event):
             if self.is_completion_widget_visible() and text:
                 self.completion_text += text
 
+    def handle_close_parentheses(self, text):
+        if not self.close_parentheses_enabled:
+            return
+        position = self.get_position('cursor')
+        rest = self.get_text('cursor', 'eol').rstrip()
+        if (len(rest) == 0 or rest[0] in (',', ')', ']', '}')):

```

Comment:
`not rest` is more pythonic than `len(rest) == 0`.
Also the surrounding parenthesis are not necessary.